cmd/admin-storage-class-add.go

```diff
@@ -0,0 +1,296 @@
+	allFlags = append(allFlags, s3OnlyFlags...)
+	allFlags = append(allFlags, azOnlyFlags...)
+	allFlags = append(allFlags, gcsOnlyFlags...)
+	return allFlags
+}
+
+var adminStorageClassAddCmd = cli.Command{
+	Name:         "add",
+	Usage:        "add a new storage class",
+	Action:       mainAdminStorageClassAdd,
+	OnUsageError: scAddUsageError,
+	Before:       setGlobalsFromContext,
+	Flags:        getAdminStorageClassAddFlags(),
+	CustomHelpTemplate: `NAME:
+  {{.HelpName}} - {{.Usage}}
+
+USAGE:
+  {{.HelpName}} TARGET NAME --type CLOUD_BACKEND [CLOUD_BACKEND_FLAGS]
+
+NAME:
+  Name of storage class. e.g s3-infrequent
+
+CLOUD_BACKEND:
+  Cloud storage backend where objects will be transitioned to specified by bucket lifecycle configuration.
+  Supported values are s3, azure or gcs.
+
+CLOUD_BACKEND_FLAGS:
+  Cloud storage backend specific flags.
+
+FLAGS:
+  {{range .VisibleFlags}}{{.}}
+  {{end}}
+EXAMPLES:
+  1. Configure a new storage class backed by AWS s3 using its Standard-IA storage class
+     {{.Prompt}} {{.HelpName}} myminio s3-infrequent --type s3 --endpoint https://s3.amazonaws.com --access-key foobar --secret-key foobar123 --region us-east-1 --bucket testbucket --prefix testprefix/ --storage-class "S3 Standard-IA"
+`,
+}
+
+// checkAdminStorageClassAddSyntax - validate all the passed arguments
+func checkAdminStorageClassAddSyntax(ctx *cli.Context) {
+	argsNr := len(ctx.Args())
+	if argsNr < 2 {
+		cli.ShowCommandHelpAndExit(ctx, ctx.Command.Name, 1) // last argument is exit code
+	}
+	if argsNr > 2 {
+		fatalIf(errInvalidArgument().Trace(ctx.Args().Tail()...),
+			"Incorrect number of arguments for sc add command.")
+	}
+}
+
+func fetchStorageClassConfig(ctx *cli.Context, scName string, scType madmin.StorageClassType) madmin.StorageClassConfig {
+	switch scType {
+	case madmin.S3:
+		accessKey := ctx.String("access-key")
+		secretKey := ctx.String("secret-key")
+		if accessKey == "" || secretKey == "" {
+			fatalIf(errInvalidArgument().Trace(), fmt.Sprintf("%s backend requires access credentials", scType))
+		}
+
+		bucket := ctx.String("bucket")
+		if bucket == "" {
+			fatalIf(errInvalidArgument().Trace(), fmt.Sprintf("%s backend requires target bucket", scType))
+		}
+
+		s3Opts := []madmin.S3Options{}
+		prefix := ctx.String("prefix")
+		if prefix != "" {
+			s3Opts = append(s3Opts, madmin.S3Prefix(prefix))
+		}
+
+		endpoint := ctx.String("endpoint")
+		if endpoint != "" {
+			s3Opts = append(s3Opts, madmin.S3Endpoint(endpoint))
+		}
+
+		region := ctx.String("region")
+		if region != "" {
+			s3Opts = append(s3Opts, madmin.S3Region(region))
+		}
+
+		s3SC := ctx.String("storage-class")
+		if s3SC != "" {
+			s3Opts = append(s3Opts, madmin.S3StorageClass(s3SC))
+		}
+
+		s3Cfg, err := madmin.NewTransitionStorageClassS3(scName, accessKey, secretKey, bucket, s3Opts...)
+		if err != nil {
+			fatalIf(probe.NewError(err), "Invalid configuration for s3 backed storage-class")
```

Comment:
```suggestion
			fatalIf(probe.NewError(err), "Invalid configuration for s3 transition target")
```

Reply:
`s3 backend storage-class` will be misleading and get confused with actual storage class used on s3